FrontEnd/Brewerweb/js/CheckFields.js

```diff
@@ -0,0 +1,127 @@
+/* Justin Evans
+*/
+// This script validates a form.
+
+// Function called when the form is submitted.
+// Function validates the form data.
+function validateForm(e) {
+    'use strict';
+
+    // Get the event object:
+	if (typeof e == 'undefined') e = window.event;
+
+    // Get form references:
+	var firstName = U.$('firstName');
+	var lastName = U.$('lastName');
+	var email = U.$('email');
+	var phone = U.$('phone');
+	var city = U.$('city');
+	var state = U.$('state');
+	var zip = U.$('zip');
+	var terms = U.$('terms');
+
+	// Flag variable:
+	var error = false;
+
+	// Validate the first name:
+	if (/^[A-Z \.\-']{2,20}$/i.test(firstName.value)) {
+		removeErrorMessage('firstName');
+	} else {
+		addErrorMessage('firstName', 'Please enter your first name.');
+		error = true;
+	}
+	//validate the last name bullet 13
+	if (/^[A-Z \.\-']{2,20}$/i.test(lastName.value)) {
+		removeErrorMessage('lastName');
+	} else {
+		addErrorMessage('lastName', 'Please enter your last name.');
+		error = true;
+	}
+	// Validate the email address:
+	if (/^[\w.-]+@[\w.-]+\.[A-Za-z]{2,6}$/.test(email.value)) {
+		removeErrorMessage('email');
+	} else {
+		addErrorMessage('email', 'Please enter your email address.');
+		error = true;
+	}
+	
+	// Validate the phone number:
+	if (/\d{3}[ \-\.]?\d{3}[ \-\.]?\d{4}/.test(phone.value)) {
+		removeErrorMessage('phone');
+	} else {
+		addErrorMessage('phone', 'Please enter your phone number.');
+		error = true;
+	}
+	//validate the city bullet 13
+	if (/^[A-Z \.\-']{2,20}$/i.test(city.value)) {
+		removeErrorMessage('city');
+	} else {
+		addErrorMessage('city', 'Please enter your citys name.');
+		error = true;
+	}
+	
+	// Validate the state:
+	if (state.selectedIndex != 0) {
+		removeErrorMessage('state');
+	} else {
+		addErrorMessage('state', 'Please select your state.');
+		error = true;
+	}
+	
+	// Validate the zip code:
+	if (/^\d{5}(-\d{4})?$/.test(zip.value)) {
+		removeErrorMessage('zip');
+	} else {
+			addErrorMessage('zip', 'Please enter your zip code.');
+		error = true;
+	}
+
+    // If an error occurred, prevent the default behavior:
+	if (error) {
+		
+
+		// Prevent the form's submission:
+	    if (e.preventDefault) {
+	        e.preventDefault();
+	    } else {
+	        e.returnValue = false;
+	    }
+	    return false;
+    
+	}
+    //window.onerror=validateForm();
+} // End of validateForm() function.
+
+// Function called when the terms checkbox changes.
+// Function enables and disables the submit button.
+function toggleSubmit() {
+	'use strict';
+    
+	// Get a reference to the submit button:
+	var submit = U.$('submit');
+	// Toggle its disabled property:
+	if (U.$('terms').checked) {
+		submit.disabled = false;
+	} else {
+		submit.disabled = true;
+	}
+	
+} // End of toggleSubmit() function.

```

Comment:
I wouldn't include and end of foo() function comment. This should be apparent by the }.

Reply:
It makes it easier to follow since many in the group are not familiar with js and there are many languages whose brackets may not necessary mean the end of a function.. ie python